### PR TITLE
[FEATURE] Language Select in Module2 - Internal Page

### PR DESCRIPTION
**Pull request by ruudsilvrants**
Add possibility to select language for a newsletter and
for receiver groups.

Added TsConfig option (langParams) to map language
variable with sys_language_uid.

Automatic falback to: &L=sys_language_uid

Example:
mod.web_modules.dmail {
   langParams.0 =
   langParams.1 = &L=1
}

In the last stap only receiver groups are shown that match
language selection. When only 1 group present select is not
shown but the group is auto selected.

Rebase of https://review.typo3.org/#/c/26378/

### CODE REVIEW

**Comment by kartolo** [comment]
Hi,

thanks for rebasing it and made a PR here. Could you:

- write a documention for this feature
- explain what this do, and how to test.

Regards,

Ivan

**Comment by ruudsilvrants** [comment]
Hello Ivan, 
I've updated the PR and here are the other request: 

- write a documention for this feature
	 - See update PR
- explain what this do
	- Explained in the first paragraph of the added documention	
- ,and how to test.
	- Create a translation for a page you want to send by Direct Mail.
	- In the BE you can see for every translation an extra icon for creating the mail, viewing the html and viewing the plain text. (indicated by the language flag over the original icon)
	- Now you can create/view the mail like a usual mail
	- In the last step you can see the receivers for that specific language (adjusting the language for a recipient list is explained in the added documentation)

If there are other remarks to this PR, please let me know.
Kind regards,
Ruud Silvrants

File: Classes/Module/Dmail.php [hunk 1]
```diff
@@ -886,19 +893,38 @@ public function cmd_finalmail()
...
         }
         $GLOBALS['TYPO3_DB']->sql_free_result($res);
 
         // added disabled. see hook
-        $select = '<select multiple="multiple" name="mailgroup_uid[]" ' . ($hookSelectDisabled ? 'disabled' : '') . '>' . implode(LF, $opt) . '</select>';
-
+        if (count($opt) === 0) {
+            /** @var $flashMessage FlashMessage */
+            $flashMessage = GeneralUtility::makeInstance('TYPO3\\CMS\\Core\\Messaging\\FlashMessage',
+                'No recipient groups found',
```
Comment (kartolo) [inline Classes/Module/Dmail.php]: this could be put in locallang file

File: ext_tables.php [hunk 2]
```diff
@@ -159,3 +159,26 @@
...
+
+/** @var \TYPO3\CMS\Core\Imaging\IconRegistry $iconRegistry */
+$iconRegistry = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance(
+    \TYPO3\CMS\Core\Imaging\IconRegistry::class
+);
+
+$iconRegistry->registerIcon(
+    'direct_mail_newmail',
+    \TYPO3\CMS\Core\Imaging\IconProvider\BitmapIconProvider::class,
+    ['source' => 'EXT:' . $_EXTKEY . '/res/gfx/newmail.gif']
```
Comment (kartolo) [inline ext_tables.php]: please change the path to /Resources/Public/Icons/

File: ext_tables.php [hunk 3]
```diff
@@ -159,3 +159,26 @@
...
+);
+
+$iconRegistry->registerIcon(
+    'direct_mail_preview_html',
+    \TYPO3\CMS\Core\Imaging\IconProvider\BitmapIconProvider::class,
+    ['source' => 'EXT:' . $_EXTKEY . '/res/gfx/preview_html.gif']
```
Comment (kartolo) [inline ext_tables.php]: this too. please change the path to /Resources/Public/Icons/

File: ext_tables.php [hunk 4]
```diff
@@ -159,3 +159,26 @@
...
+);
+
+$iconRegistry->registerIcon(
+    'direct_mail_preview_plain',
+    \TYPO3\CMS\Core\Imaging\IconProvider\BitmapIconProvider::class,
+    ['source' => 'EXT:' . $_EXTKEY . '/res/gfx/preview_txt.gif']
```
Comment (kartolo) [inline ext_tables.php]: please change the path to /Resources/Public/Icons/

**Pull request merged**